test/lib/ufe/testUngroupCmd.py

```diff
@@ -0,0 +1,415 @@
+        translateAttr = spherePrim.GetAttribute('xformOp:translate')
+        self.assertEqual(translateAttr.Get(), Gf.Vec3d(2.0, 0.0, 0.0))
+
+        spherePrim = mayaUsd.ufe.ufePathToPrim(ufe.PathString.string(sphere2Path))
+        translateAttr = spherePrim.GetAttribute('xformOp:translate')
+        self.assertEqual(translateAttr.Get(), Gf.Vec3d(-2.0, 0.0, 0.0))
+
+    def testUngroupWorld(self):
+        '''Verify -world flag.'''
+        
+        # create a sphere generator
+        sphereGen = SphereGenerator(4, self.contextOp, self.proxyShapePathStr)
+
+        # create 4 spheres
+        pathList = list()
+        for _ in range(4):
+            pathList.append(sphereGen.createSphere())
+
+        # group /Sphere2, /Sphere3, /Sphere4
+        cmds.group(joinPathSegments(pathList[1]),
+                   joinPathSegments(pathList[2]),
+                   joinPathSegments(pathList[3]))
+
+        # verify the paths after grouping
+        assert ([x for x in self.stage.Traverse()] == 
+            [self.stage.GetPrimAtPath("/Sphere1"),
+            self.stage.GetPrimAtPath("/group1"),
+            self.stage.GetPrimAtPath("/group1/Sphere2"),
+            self.stage.GetPrimAtPath("/group1/Sphere3"),
+            self.stage.GetPrimAtPath("/group1/Sphere4")])
+
+        # group /group1/Sphere2, /group1/Sphere3, /group1/Sphere4
+        cmds.group("{},/group1/Sphere2".format(self.proxyShapePathStr),
+                   "{},/group1/Sphere3".format(self.proxyShapePathStr),
+                   "{},/group1/Sphere4".format(self.proxyShapePathStr))
+
+        # verify the paths after second grouping
+        assert ([x for x in self.stage.Traverse()] == 
+            [self.stage.GetPrimAtPath("/Sphere1"),
+            self.stage.GetPrimAtPath("/group1"),
+            self.stage.GetPrimAtPath("/group1/group1"),
+            self.stage.GetPrimAtPath("/group1/group1/Sphere2"),
+            self.stage.GetPrimAtPath("/group1/group1/Sphere3"),
+            self.stage.GetPrimAtPath("/group1/group1/Sphere4")])
+
+        # remove /group1/group1 from the hierarchy with the -world flag
+        cmds.ungroup("{},/group1/group1".format(self.proxyShapePathStr), world=True)
+
+        # verify the paths after ungroup
+        assert ([x for x in self.stage.Traverse()] == 
+            [self.stage.GetPrimAtPath("/Sphere1"),
+            self.stage.GetPrimAtPath("/group1"),
+            self.stage.GetPrimAtPath("/Sphere2"),
+            self.stage.GetPrimAtPath("/Sphere3"),
+            self.stage.GetPrimAtPath("/Sphere4")])
+
+    @unittest.skip("parent flag is not supported yet")
+    def testUngroupParent(self):
+        '''Verify -parent flag.'''
+        pass
+
+    def testUngroupProxyShape(self):
+        '''Verify ungrouping of the proxyShape.'''
+
+        # select proxyShape
+        cmds.select(ufe.PathString.string(self.proxyShapeItem.path()))
+
+        # create a group
+        cmds.group()
```

Comment:
This test is grouping the proxy shape, then ungrouping.  Not sure how interesting this is: it's pretty standard Maya functionality.

What I would like to confirm is that trying to ungroup the proxy shape itself (which groups the pseudo-root's children) is not possible.  If it's prevented by virtue of the proxy shape being a shape, and thus incapable of being a parent to other Maya nodes, that's enough for me --- you can adjust this test to demonstrate that.